src/components/SignupForm/index.js

```diff
@@ -1,100 +1,51 @@
 import React from 'react';
-import { connect } from 'react-redux';
-import {
-  usernameChanged,
-  emailChanged,
-  passwordChanged,
-  confPasswordChanged,
-  signUpUser
-} from '../../actions/signupSigninActions';
 import { Link } from 'react-router-dom';
 import InputBox from '../InputBox';
 import Button from '../SignupButton';
 
 const Signup = props => {
-  const { signupData } = props;
-
-  const onChangeUsername = e => {
-    props.usernameChanged(e.target.value);
-  };
-
-  const onChangeEmail = e => {
-    props.emailChanged(e.target.value);
-  };
-
-  const onChangePassword = e => {
-    props.passwordChanged(e.target.value);
-  };
-
-  const onChangeConfPassword = e => {
-    props.confPasswordChanged(e.target.value);
-  };
-
-  const handleSubmit = e => {
-    e.preventDefault();
-    props.signUpUser(signupData);
-  };
-
   return (
     <div>
       <div className="signStr spacebottom">Create an account</div>
-      <form onSubmit={handleSubmit}>
+      <form onSubmit={props.handleSubmit}>
         <InputBox
           type="text"
-          value={signupData.username}
+          value={props.username}
           name="username"
           placeholder="USERNAME"
-          onChangeInput={e => onChangeUsername(e)}
+          onChangeInput={props.onChange}
         />
         <InputBox
           type="email"
-          value={signupData.email}
+          value={props.email}
           name="email"
           placeholder="EMAIL"
-          onChangeInput={e => onChangeEmail(e)}
+          onChangeInput={props.onChange}
         />
         <InputBox
           type="password"
-          value={signupData.password}
+          value={props.password}
           name="password"
           placeholder="PASSWORD"
-          onChangeInput={e => onChangePassword(e)}
+          onChangeInput={props.onChange}
         />
         <InputBox
           type="password"
-          value={signupData.confirm_password}
-          name="confirm-password"
+          value={props.confirm_password}
+          name="confirm_password"
           placeholder="CONFIRM PASSWORD"
-          onChangeInput={e => onChangeConfPassword(e)}
+          onChangeInput={props.onChange}
         />
-        {signupData.error !== '' ? (
-          <div className="error-msg">{signupData.error}</div>
-        ) : null}
+        {props.error && <div className="error-msg">{props.error}</div>}
         <Button value="Signup" />
         <div className="other">
           <span>
-            Already have an account?{' '}
-            <Link to={'/login'}>Signin</Link>
+            Already have an account? <Link to={'/login'}>Signin</Link>
```

Comment:
is `/login` a prop?

Reply:
Yes @hadijahkyampeire, It is a prop of `Link` to represent where the page should link to

Reply:
👍